File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/TestResponseDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.TestStatus;
import com.example.oriengo.model.enumeration.TestType;
import lombok.*;
import org.springframework.web.bind.annotation.CrossOrigin;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TestResponseDTO {

    private Long id;
    private Long studentId;
    private TestType type;
    private TestStatus status;
    private LocalDateTime startedAt;
    private LocalDateTime updatedAt;
    private Integer answeredQuestionsCount;
    private LocalDateTime completedAt;
    private Integer durationMinutes;
    private Integer questionsCount;
}

File: OrienGO-Frontend/src/app/model/dto/TestResponseDTO.ts
// src/app/model/dto/TestResponseDTO.ts
import { TestType } from '../enum/TestType';
import { TestStatus } from '../enum/TestStatus';

export interface TestResponseDTO {
  id: number;
  studentId: number;
  type: TestType;
  status: TestStatus;
  startedAt: string;       // ISO date string from backend
  completedAt: string;     // ISO date string from backend
  updatedAt: string;     // ISO date string from backend
  durationMinutes: number;
  questionsCount: number;
  answeredQuestionsCount: number;
  softDeleted: boolean;
}
